src/main/kotlin/org/rust/ide/newProject/ui/RsNewProjectPanel.kt

```diff
@@ -5,52 +5,147 @@
+import com.intellij.ui.ColoredListCellRenderer
+import com.intellij.ui.SimpleTextAttributes
+import com.intellij.ui.components.*
 import org.rust.cargo.project.settings.ui.RustProjectSettingsPanel
+import org.rust.cargo.toolchain.Cargo
 import org.rust.ide.newProject.ConfigurationData
+import org.rust.ide.newProject.RsCustomTemplate
+import org.rust.ide.newProject.RsGenericTemplate
+import org.rust.ide.newProject.RsProjectTemplate
 import org.rust.ide.ui.RsLayoutBuilder
-import java.awt.event.ItemEvent
-import java.awt.event.ItemListener
-import javax.swing.ButtonGroup
+import org.rust.openapiext.GeneralCommandLine
+import org.rust.openapiext.UiDebouncer
+import org.rust.openapiext.execute
+import javax.swing.JList
+import javax.swing.ListSelectionModel
+import javax.swing.ScrollPaneConstants
 
 class RsNewProjectPanel(
-    private val showProjectTypeCheckbox: Boolean,
-    updateListener: (() -> Unit)? = null
+    private val showProjectTypeSelection: Boolean,
+    private val updateListener: (() -> Unit)? = null
 ) : Disposable {
 
     private val rustProjectSettings = RustProjectSettingsPanel(updateListener = updateListener)
-    private val templateButtons = listOf(JBRadioButton("Binary (application)", true), JBRadioButton("Library"))
-    private val checkboxListener = ItemListener {
-        if (it.stateChange == ItemEvent.SELECTED) updateListener?.invoke()
-    }
 
-    init {
-        templateButtons.forEach { it.addItemListener(checkboxListener) }
+    private val templateList = JBList(
+        RsGenericTemplate("Binary (application)", true),
+        RsGenericTemplate("Library", false),
+        RsCustomTemplate("WebAssembly Lib", "https://github.com/rustwasm/wasm-pack-template")
+    ).apply {
+        selectionMode = ListSelectionModel.SINGLE_SELECTION
+        selectedIndex = 0
+        addListSelectionListener { update() }
+        cellRenderer = object : ColoredListCellRenderer<RsProjectTemplate>() {
+            override fun customizeCellRenderer(list: JList<out RsProjectTemplate>,
+                                               value: RsProjectTemplate?,
+                                               index: Int,
+                                               selected: Boolean,
+                                               hasFocus: Boolean) {
+                if (value == null) return
+
+                icon = value.icon
+                append(value.name)
+
+                if (value is RsCustomTemplate) {
+                    append(" ")
+                    append(value.shortLink, SimpleTextAttributes.GRAYED_ATTRIBUTES)
+                }
+            }
+        }
+    }
+    private val templatePane = JBScrollPane(templateList).apply {
+        horizontalScrollBarPolicy = ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER
     }
 
-    val data: ConfigurationData get() = ConfigurationData(rustProjectSettings.data, templateButtons[0].isSelected)
+    private var isCargoGenerateInstalled = false
```

Comment:
AFAIU it is needed only to download `cargo-generate` if it is really needed. So let's rename to something like `needInstallCargoGenerate` and use `false` in case of `RsGenericTemplate`